File: desktopclient/src/main/java/com/goexp/galgame/gui/view/common/jump/JumpLinkController.java
package com.goexp.galgame.gui.view.common.jump;

import com.goexp.galgame.common.website.GetchuURL;
import com.goexp.galgame.gui.model.Game;
import com.goexp.galgame.gui.util.Websites;
import javafx.fxml.FXML;
import javafx.scene.control.MenuItem;


public class JumpLinkController {

    private Game game = new Game();

    @FXML
    private SearchController searchLinkController;

    @FXML
    private MenuItem linkGetchu;


    @FXML
    private void initialize() {

        linkGetchu.setOnAction((e) -> {
            Websites.open(GetchuURL.Game.byId(game.id));
        });
    }

    public void load(Game game) {
        this.game = game;

        searchLinkController.load(game.name.split("[\\s～\\-]")[0]);

    }


}

File: desktopclient/src/main/scala/com/goexp/galgame/gui/view/common/jump/JumpLinkController.scala
package com.goexp.galgame.gui.view.common.jump

import com.goexp.galgame.gui.HGameApp
import com.goexp.ui.javafx.DefaultController
import javafx.beans.property.{SimpleIntegerProperty, SimpleStringProperty}
import javafx.fxml.FXML
import javafx.scene.control.MenuItem
import scalafx.Includes._

class JumpLinkController extends DefaultController {

  final lazy val name = new SimpleStringProperty()
  final lazy val id = new SimpleIntegerProperty()

  //  @FXML private var searchLinkController: SearchController = _
  //  @FXML private var linkGetchu: MenuItem = _
  @FXML private var linkGuide: MenuItem = _

  override protected def eventBinding() = {
    //    linkGetchu.setOnAction(_ => Websites.open(GetchuGameRemote.byId(id.value)))
    linkGuide.setOnAction(_ => HGameApp.loadGuide(name.value))
  }

}